libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -305,16 +305,18 @@ func (r *reporter) snapshotLoop(namespace, prefix string, period time.Duration)
 			"params": map[string]string{"interval": strconv.Itoa(int(period/time.Second)) + "s"},
 		}
 
-		clusterUUID := getClusterUUID()
+		clusterUUID, okToPublish := getClusterUUID()
 		if clusterUUID != "" {
 			meta.Put("cluster_uuid", clusterUUID)
 		}
 
-		r.client.Publish(beat.Event{
-			Timestamp: ts,
-			Fields:    fields,
-			Meta:      meta,
-		})
+		if okToPublish {
+			r.client.Publish(beat.Event{
+				Timestamp: ts,
+				Fields:    fields,
+				Meta:      meta,
+			})
+		}
```

Comment:
Are we informing the user of the fact that we have decided not to publish an event? Perhaps it would be good to log if we are not OK to publish?

Reply:
I don't think we should spam the log with this information at the `INFO` level but it might be good to have at the `DEBUG` level?

Reply:
+1 for debug. Connections in Beats are lazy. If there is no event to be published, then we will never connect.
Maybe it's worth to change the 'connection' behavior. Still have it deferred (to not block on startup), but connect asynchronously in the background, even if there is no data. Otherwise users might wonder why the Beat doesn't appear in the monitoring UI, although it's running.

Reply:
> Maybe it's worth to change the 'connection' behavior. Still have it deferred (to not block on startup), but connect asynchronously in the background, even if there is no data. Otherwise users might wonder why the Beat doesn't appear in the monitoring UI, although it's running.

Do you think this merits some discussion first with the rest of the Beats team? If so I can file a `discuss` issue about it. If not, I can put up a PR implementing the change. Which one would you suggest?

Reply:
Maybe we want to discuss it first. See my other comment about hooking up monitoring + outputs. Having outputs available directly in monitoring would allow us to force the connection setup via monitoring as well. The first attempt to get any information (or publish) from/to ES would result in a connection attempt.

Reply:
++ let me first look into what it takes to get outputs into the monitoring code scope as that would solve the connection issue as well as the cluster UUID lookup issue more robustly.